src/main/java/org/sonar/ucfg/UCFGtoProtobuf.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * sonar-ucfg
+ * Copyright (C) 2018-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.ucfg;
+
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.sonar.ucfg.protobuf.Ucfg;
+
+public final class UCFGtoProtobuf {
+
+  private UCFGtoProtobuf() {
+    // empty constructor
+  }
+
+  public static void toProtobufFile(UCFG ucfg, String filename) throws IOException {
+    Ucfg.UCFG protobufUCFG = Ucfg.UCFG.newBuilder()
+      .setMethodId(ucfg.methodId())
+      .setLocation(toProtobuf(ucfg.location()))
+      .addAllParameters(ucfg.parameters().stream().map(Expression.Variable::id).collect(Collectors.toList()))
+      .addAllUcfgEntries(ucfg.entryBlocks().stream().map(BasicBlock::label).map(Label::id).collect(Collectors.toList()))
+      .addAllBasicBlocks(ucfg.basicBlocks().values().stream().map(UCFGtoProtobuf::toProtobuf).collect(Collectors.toList()))
+      .build();
+    protobufUCFG.writeTo(new FileOutputStream(filename));
+
+  }
+
+  private static Ucfg.BasicBlock toProtobuf(BasicBlock basicBlock) {
+    Ucfg.BasicBlock.Builder builder = Ucfg.BasicBlock.newBuilder()
+      .setId(basicBlock.label().id())
+      .setLocation(toProtobuf(basicBlock.locationInFile()))
+      .addAllInstruction(basicBlock.calls().stream().map(UCFGtoProtobuf::toProtobuf).collect(Collectors.toList()));
```

Comment:
@benzonico I wonder why `addAllInstruction` in singular

Reply:
This is generated code and comes from protobuf file. Fixed in there.